sonaranalyzer-dotnet/its/create-issue-reports.ps1

```diff
@@ -142,13 +143,23 @@ function New-IssueReports([string]$sarifReportPath) {
         $object = New-Object –Type System.Object
         $object | Add-Member –Type NoteProperty –Name issues –Value $_.Group
 
-        $path = Join-Path (Join-Path 'actual' $project) ($file.BaseName + '-' + $_.Name + $file.Extension)
+        $actualProjectFolder = Join-Path 'actual' $project
+        $issueFileName = $file.BaseName + '-' + $_.Name + $file.Extension
+        $path = Join-Path $actualProjectFolder $issueFileName
+
         $lines =
             (
                 (ConvertTo-Json $object -Depth 42 |
                     ForEach-Object { [System.Text.RegularExpressions.Regex]::Unescape($_) }  # Unescape powershell to json automatic escape
-                ) -split "`r`n"										            	         # Convert JSON to String and split lines
+                ) -split "`r`n"                                                              # Convert JSON to String and split lines
             ) | Foreach-Object { $_.TrimStart() }                                            # Remove leading spaces
-        Set-Content $path $lines
+
+        if (Test-Path $path) {
+            Set-Content $path $lines
+        } else {
+            Write-Host "The path '${path}' does not exist, will first create the file."
```

Comment:
This is a nice change but do you know when the path does not exist? It's a bit strange that this didn't occurred until now.

Reply:
Yes, it happens when adding a new test project:
- when adding a new test project, the "expected" folder doesn't exist yet
- the `actual` folder is created in [function Initialize-ActualFolder()](https://github.com/SonarSource/sonar-dotnet/blob/a11a3dca45803ae093705ceb5de90cc28af36468/sonaranalyzer-dotnet/its/regression-test.ps1#L89) where it gets initialized with the contents of the "expected" folder
- after running the test, the existing files are overwritten with `Set-Content $path $lines` - if the `$path` does not exist, it fails. And it doesn't exist, because the "expected" folder doesn't exist, so no files could be copied.

Reply:
Actually inside `Initialize-ActualFolder` I see `# this copies no files if ruleId is not set, and all but ending with ruleId if set`.

And I tested now, and `Initialize-ActualFolder` only creates empty folders if the tests are running for all rules.

And then here in `New-IssueReports` the files get created. So here, we only need to make sure that the folder exists (because it's not created in `Initialize-ActualFolder`), and if it doesn't exist, create it.